android/app/src/main/java/updated/mysterium/vpn/ui/manual/connect/home/HomeViewModel.kt

```diff
@@ -1,14 +1,150 @@
 package updated.mysterium.vpn.ui.manual.connect.home
 
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.launch
+import mysterium.ConnectRequest
+import network.mysterium.AppNotificationManager
+import network.mysterium.proposal.ProposalViewItem
+import network.mysterium.service.core.DeferredNode
+import network.mysterium.service.core.MysteriumCoreService
+import network.mysterium.service.core.Statistics
+import network.mysterium.wallet.IdentityModel
+import network.mysterium.wallet.IdentityRegistrationStatus
 import updated.mysterium.vpn.common.extensions.liveDataResult
+import updated.mysterium.vpn.model.manual.connect.ProposalModel
 import updated.mysterium.vpn.network.provider.usecase.UseCaseProvider
 
 class HomeViewModel(useCaseProvider: UseCaseProvider) : ViewModel() {
 
+    private val _statisticsUpdate = MutableLiveData<Statistics>()
+    val statisticsUpdate
+        get() = _statisticsUpdate
+
+    private val _connectionState = MutableLiveData<String>()
+    val connectionState
+        get() = _connectionState
```

Comment:
Can you explain the purpose of not making it just with one public variable?

Reply:
I did this to improve security. It can be simplified this way:
var statisticsUpdate = MutableLiveData<Statistics>()
        private set

Reply:
Fix